app/views/courses/show.html.haml

```diff
@@ -13,10 +13,10 @@
           %p
             %b= semester
             \:
-          - instances.each do |instance|
+          - instances.sort_by { |instance| instance.section }.each do |instance|
             Section #{instance.section} taught by #{instance.professor.name}
             %br
-            = instance.meetings.first.schedule
+            = instance.meetings.try(:first).try(:schedule) || 'TBA'
```

Comment:
Yeah I think you're right. So this actually becomes a one-liner: `meetings.map(&:schedule).join('; ') || 'TBA'`